src/sgame/sg_spawn_mover.cpp

```diff
@@ -2449,34 +2449,44 @@ void SP_func_dynamic( gentity_t *self )
 
 ROTATING
 
+QUAKED func_rotating (0 .5 .8) ? START_ON - X_AXIS Y_AXIS
```

Comment:
What does this mean?

Reply:
I added the comment from tremulous (or older) back to it, which explained all settings and usage in netradiant.
With them its easier to understand and also clear what the default values and expected behaviors are.

Reply:
I wonder if at some point in the past QERadiant (QE for Quake Ed) was reading the entity definitions from comments in source code, looking for `/*QUAKED` patterns.

Here being how this entity keyword being defined in QRRadiant .def format:

https://github.com/Unvanquished/unvanquished-mapeditor-support/blob/c49616ca9b0548511993c2d830b3524549bc26ae/build/gtkradiant/install/pkg/scripts/entities.def#L407-L429

```c
/*QUAKED func_rotating (0 0.498 0.8) - - X_AXIS Y_AXIS
------ FLAGS ------
X_AXIS: Entity will rotate around the X axis.
Y_AXIS: Entity will rotate around the Y axis.
------ PROPERTIES ------
origin: [required] Rotation axis position. (vec3_float)
speed: [required] Determines how fast entity rotates. (float, def: 400)
color: Colour of emitted light. Emission activated by either color or light set. Don't forget to set origin. (vec3_float (color), def: 1 1 1)
dmg: Damage dealt when blocked. (int, def: 2)
group: Name of entity group. Allows to activate all entities of group at once.
light: Light emitting intensity. Emission activated by either color or light set. Don't forget to set origin. (float, def: 100)
model2: Path to custom model attached to entity. Don't forget to set "origin". (string (model))
noise: Sound played when activated via targetname. Emitted at origin. (string (sound))
targetname: Doesn't work properly. (string (targetname))
_celshader: [q3map] Sets the cel shader used for this geometry. Note: omit the "textures/" prefix. Example: http://q3map2.robotrenegade.com/docs/shader_manual/cel-shading.html (string, eg: mymap/ink)
_cs: [q3map] or "_castShadows". Group number entity cast shadow to. Negative value uses same group but disables world. 0 disables shadow casting. (int)
_lightmapscale: [q3map] Value scaling the resolution of lightmaps. Lesser value gives bigger resolution. (float, def: 1)
_rs: [q3map] or "_receiveShadows". Group number entity receive shadow from. Negative value uses same group but disables world. 0 disables shadow receiving. (int)
------ DESCRIPTION ------
Solid entity that rotates continuously. Rotates on the Z axis by default.
For "_celshader" read http://q3map2.robotrenegade.com/docs/shader_manual/cel-shading.html
Entity supports [terrain] properties. They're not so useful here, but you can refer to func_group for descriptions.
*/
```

Reply:
ioq3 is still full of this kind of comments:

```
$ rg QUAKED ~/dev/ioq3
/home/illwieckz/dev/ioq3/code/game/bg_misc.c
28:/*QUAKED item_***** ( 0 0 0 ) (-16 -16 -16) (16 16 16) suspended
65:/*QUAKED item_armor_shard (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
82:/*QUAKED item_armor_combat (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
98:/*QUAKED item_armor_body (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
117:/*QUAKED item_health_small (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
134:/*QUAKED item_health (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
151:/*QUAKED item_health_large (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
168:/*QUAKED item_health_mega (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
190:/*QUAKED weapon_gauntlet (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
206:/*QUAKED weapon_shotgun (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
222:/*QUAKED weapon_machinegun (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
238:/*QUAKED weapon_grenadelauncher (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
254:/*QUAKED weapon_rocketlauncher (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
270:/*QUAKED weapon_lightning (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
286:/*QUAKED weapon_railgun (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
302:/*QUAKED weapon_plasmagun (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
318:/*QUAKED weapon_bfg (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
334:/*QUAKED weapon_grapplinghook (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
354:/*QUAKED ammo_shells (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
370:/*QUAKED ammo_bullets (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
386:/*QUAKED ammo_grenades (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
402:/*QUAKED ammo_cells (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
418:/*QUAKED ammo_lightning (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
434:/*QUAKED ammo_rockets (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
450:/*QUAKED ammo_slugs (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
466:/*QUAKED ammo_bfg (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
485:/*QUAKED holdable_teleporter (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
500:/*QUAKED holdable_medkit (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
521:/*QUAKED item_quad (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
538:/*QUAKED item_enviro (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
555:/*QUAKED item_haste (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
572:/*QUAKED item_invis (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
589:/*QUAKED item_regen (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
606:/*QUAKED item_flight (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
623:/*QUAKED team_CTF_redflag (1 0 0) (-16 -16 -16) (16 16 16)
640:/*QUAKED team_CTF_blueflag (0 0 1) (-16 -16 -16) (16 16 16)
658:/*QUAKED holdable_kamikaze (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
674:/*QUAKED holdable_portal (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
690:/*QUAKED holdable_invulnerability (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
706:/*QUAKED ammo_nails (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
722:/*QUAKED ammo_mines (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
738:/*QUAKED ammo_belt (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
757:/*QUAKED item_scout (.3 .3 1) (-16 -16 -16) (16 16 16) suspended redTeam blueTeam
773:/*QUAKED item_guard (.3 .3 1) (-16 -16 -16) (16 16 16) suspended redTeam blueTeam
789:/*QUAKED item_doubler (.3 .3 1) (-16 -16 -16) (16 16 16) suspended redTeam blueTeam
805:/*QUAKED item_doubler (.3 .3 1) (-16 -16 -16) (16 16 16) suspended redTeam blueTeam
821:	/*QUAKED team_CTF_neutralflag (0 0 1) (-16 -16 -16) (16 16 16)
865:/*QUAKED weapon_nailgun (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
881:/*QUAKED weapon_prox_launcher (.3 .3 1) (-16 -16 -16) (16 16 16) suspended
902:/*QUAKED weapon_chaingun (.3 .3 1) (-16 -16 -16) (16 16 16) suspended

/home/illwieckz/dev/ioq3/code/game/g_client.c
30:/*QUAKED info_player_deathmatch (1 0 1) (-16 -16 -24) (16 16 32) initial
50:/*QUAKED info_player_start (1 0 0) (-16 -16 -24) (16 16 32)
58:/*QUAKED info_player_intermission (1 0 1) (-16 -16 -24) (16 16 32)

/home/illwieckz/dev/ioq3/code/game/g_misc.c
28:/*QUAKED func_group (0 0 0) ?
33:/*QUAKED info_camp (0 0.5 0) (-4 -4 -4) (4 4 4)
41:/*QUAKED info_null (0 0.5 0) (-4 -4 -4) (4 4 4)
49:/*QUAKED info_notnull (0 0.5 0) (-4 -4 -4) (4 4 4)
58:/*QUAKED light (0 1 0) (-8 -8 -8) (8 8 8) linear
127:/*QUAKED misc_teleporter_dest (1 0 0) (-32 -32 -24) (32 32 -16)
138:/*QUAKED misc_model (1 0 0) (-16 -16 -16) (16 16 16)
204:/*QUAKED misc_portal_surface (0 0 1) (-8 -8 -8) (8 8 8)
224:/*QUAKED misc_portal_camera (0 0 1) (-8 -8 -8) (8 8 8) slowrotate fastrotate noswing
315:/*QUAKED shooter_rocket (1 0 0) (-16 -16 -16) (16 16 16)
323:/*QUAKED shooter_plasma (1 0 0) (-16 -16 -16) (16 16 16)
331:/*QUAKED shooter_grenade (1 0 0) (-16 -16 -16) (16 16 16)

/home/illwieckz/dev/ioq3/code/game/g_mover.c
932:/*QUAKED func_door (0 .5 .8) ? START_OPEN x CRUSHER
1107:/*QUAKED func_plat (0 .5 .8) ?
1187:/*QUAKED func_button (0 .5 .8) ?
1398:/*QUAKED path_corner (.5 .3 0) (-8 -8 -8) (8 8 8)
1415:/*QUAKED func_train (0 .5 .8) ? START_ON TOGGLE BLOCK_STOPS
1468:/*QUAKED func_static (0 .5 .8) ?
1491:/*QUAKED func_rotating (0 .5 .8) ? START_ON - X_AXIS Y_AXIS
1541:/*QUAKED func_bobbing (0 .5 .8) ? X_AXIS Y_AXIS
1589:/*QUAKED func_pendulum (0 .5 .8) ?

/home/illwieckz/dev/ioq3/code/game/g_target.c
27:/*QUAKED target_give (1 0 0) (-8 -8 -8) (8 8 8)
63:/*QUAKED target_remove_powerups (1 0 0) (-8 -8 -8) (8 8 8)
90:/*QUAKED target_delay (1 0 0) (-8 -8 -8) (8 8 8)
119:/*QUAKED target_score (1 0 0) (-8 -8 -8) (8 8 8)
138:/*QUAKED target_print (1 0 0) (-8 -8 -8) (8 8 8) redteam blueteam private
169:/*QUAKED target_speaker (1 0 0) (-8 -8 -8) (8 8 8) looped-on looped-off global activator
251:/*QUAKED target_laser (0 .5 .8) (-8 -8 -8) (8 8 8) START_ON
359:/*QUAKED target_teleporter (1 0 0) (-8 -8 -8) (8 8 8)
372:/*QUAKED target_relay (.5 .5 .5) (-8 -8 -8) (8 8 8) RED_ONLY BLUE_ONLY RANDOM
405:/*QUAKED target_kill (.5 .5 .5) (-8 -8 -8) (8 8 8)
416:/*QUAKED target_position (0 0.5 0) (-4 -4 -4) (4 4 4)
453:/*QUAKED target_location (0 0.5 0) (-8 -8 -8) (8 8 8)

/home/illwieckz/dev/ioq3/code/game/g_team.c
1162:/*QUAKED team_CTF_redplayer (1 0 0) (-16 -16 -16) (16 16 32)
1169:/*QUAKED team_CTF_blueplayer (0 0 1) (-16 -16 -16) (16 16 32)
1176:/*QUAKED team_CTF_redspawn (1 0 0) (-16 -16 -24) (16 16 32)
1183:/*QUAKED team_CTF_bluespawn (0 0 1) (-16 -16 -24) (16 16 32)
1381:/*QUAKED team_redobelisk (1 0 0) (-16 -16 0) (16 16 8)
1406:/*QUAKED team_blueobelisk (0 0 1) (-16 -16 0) (16 16 88)
1431:/*QUAKED team_neutralobelisk (0 0 1) (-16 -16 0) (16 16 88)

/home/illwieckz/dev/ioq3/code/game/g_spawn.c
557:/*QUAKED worldspawn (0 0 0) ?

/home/illwieckz/dev/ioq3/code/game/g_trigger.c
87:/*QUAKED trigger_multiple (.5 .5 .5) ? RED_ONLY BLUE_ONLY
125:/*QUAKED trigger_always (.5 .5 .5) (-8 -8 -8) (8 8 8)
195:/*QUAKED trigger_push (.5 .5 .5) ?
237:/*QUAKED target_push (.5 .5 .5) (-8 -8 -8) (8 8 8) bouncepad
297:/*QUAKED trigger_teleport (.5 .5 .5) ? SPECTATOR
334:/*QUAKED trigger_hurt (.5 .5 .5) ? START_OFF - SILENT NO_PROTECTION SLOW
414:/*QUAKED func_timer (0.3 0.1 0.6) (-8 -8 -8) (8 8 8) START_ON
```

It looks like we only have one left in Unvanquished:

```
rg QUAKED ~/dev/Unvanquished
/home/illwieckz/dev/Unvanquished/src/sgame/sg_spawn.cpp
1012:/*QUAKED worldspawn (0 0 0) ?
```

The whole comment says explicitely “_Warning: The following comment contains information, that might be parsed and used by radiant based mapeditors._”:

```
/**
 * Warning: The following comment contains information, that might be parsed and used by radiant based mapeditors.
 */
/*QUAKED worldspawn (0 0 0) ?
Used for game-world global options.
Every map should have exactly one.

=== KEYS ===
; message: Text to print during connection process. Used for the name of level.
; music: path/name of looping sound file used for level's music (eg. music/sonic5).
; gravity: level gravity [g_gravity (800)]

; humanBuildPoints: maximum amount of power the humans can use. [g_humanBuildPoints]
; alienBuildPoints: maximum amount of sentience available to the overmind. [g_alienBuildPoints]

; disabledEquipment: A comma delimited list of human weapons or upgrades to disable for this map. [g_disabledEquipment ()]
; disabledClasses: A comma delimited list of alien classes to disable for this map. [g_disabledClasses ()]
; disabledBuildables: A comma delimited list of buildables to disable for this map. [g_disabledBuildables ()]
*/
```

Though we don't need to take care of that `/*QUAKED` keyword, no maintained editor rely on this anymore